src/bpftrace.cpp

```diff
@@ -601,6 +601,27 @@ std::unique_ptr<AttachedProbe> BPFtrace::attach_probe(Probe &probe, const BpfOrc
   return nullptr;
 }
 
+bool attach_reverse(const Probe &p)
+{
+  switch(p.type)
+  {
+    case ProbeType::kprobe:
+    case ProbeType::uprobe:
+    case ProbeType::uretprobe:
+    case ProbeType::usdt:
+    case ProbeType::software:
+      return true;
+    case ProbeType::kretprobe:
+    case ProbeType::tracepoint:
+    case ProbeType::profile:
+    case ProbeType::interval:
+    case ProbeType::hardware:
+      return false;
+    default:
```

Comment:
If you omit the `default` section, we can move this check to compile time rather than run time. The compiler should warn us if we didn't handle every enum value. Although that being said, I've seen this generate warnings on some compilers.

Reply:
Yeah, I had it that way originally (`default` was replaced by `case ProbeType::invalid` so that all the cases were covered), but the compiler was complaining:
```
/home/ubuntu/bpftrace/src/bpftrace.cpp: In function ‘bool bpftrace::attach_reverse(const bpftrace::Probe&)’:
/home/ubuntu/bpftrace/src/bpftrace.cpp:606:9: warning: switch missing default case [-Wswitch-default]
   switch(p.type)
         ^
…
/home/ubuntu/bpftrace/src/bpftrace.cpp: In function ‘bool bpftrace::attach_reverse(const bpftrace::Probe&)’:
/home/ubuntu/bpftrace/src/bpftrace.cpp:623:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
```

Reply:
We'd only want to rely on the compiler to check this if we have `-Werror` enabled. We're almost there, but not quite yet!